kin-sdk/kin-sdk-lib/src/main/java/kin/sdk/Transaction.java

```diff
@@ -1,65 +1,61 @@
 package kin.sdk;
 
-import java.math.BigDecimal;
+import java.io.IOException;
 import kin.base.KeyPair;
+import kin.base.Memo;
+import kin.base.Network;
 
 public class Transaction {
 
-    private final KeyPair destination;
-    private final KeyPair source;
-    private final BigDecimal amount;
-    private final int fee;
-    private final String memo;
-
     /**
      * The transaction hash
      */
     private final TransactionId id;
 
-    private final kin.base.Transaction stellarTransaction;
-    private final WhitelistableTransaction whitelistableTransaction;
+    private final kin.base.Transaction baseTransaction;
 
-    Transaction(KeyPair destination, KeyPair source, BigDecimal amount, int fee, String memo,
-                TransactionId id, kin.base.Transaction stellarTransaction, WhitelistableTransaction whitelistableTransaction) {
-        this.destination = destination;
-        this.source = source;
-        this.amount = amount;
-        this.fee = fee;
-        this.memo = memo;
+    public Transaction(TransactionId id, kin.base.Transaction baseTransaction) {
         this.id = id;
-        this.stellarTransaction = stellarTransaction;
-        this.whitelistableTransaction = whitelistableTransaction;
+        this.baseTransaction = baseTransaction;
     }
 
-    public KeyPair getDestination() {
-        return destination;
+    public static Transaction decodeTransaction(String transactionEnvelope) throws IOException {
+        kin.base.Transaction transaction = kin.base.Transaction.fromEnvelopeXdr(transactionEnvelope);
+        TransactionId id = new TransactionIdImpl(Utils.byteArrayToHex(transaction.hash()));
+        return new Transaction(id, transaction);
     }
 
     public KeyPair getSource() {
-        return source;
-    }
-
-    public BigDecimal getAmount() {
-        return amount;
+        return baseTransaction.getSourceAccount();
     }
 
     public int getFee() {
-        return fee;
+        return baseTransaction.getFee();
     }
 
-    public String getMemo() {
-        return memo;
+    public Memo getMemo() {
```

Comment:
breaks the current API  (you will need to upgrade the migration module)

Reply:
This new refactor of this class will anyway breaks the API.
The change is required because this is how a Transaction object which wraps the real stellar transaction object(with our additions) should look like.
Also there were members that should not even be in a Transaction object, for example, the amount is not part of a transaction but a part of the operation.
A transaction object should only have:
operation count, addOperation, setFee, setMemo, setTimeBounds and also getters for them.
It should have the ability to sign and get an envelope or be created from an envelope.

Regarding the migration module: we probably will not add anymore changes to the migration module because in 2 weeks it should be deprecated.